workers/java/io/temporal/omes/KitchenSinkWorkflowImpl.java

```diff
@@ -0,0 +1,457 @@
+package io.temporal.omes;
+
+import com.google.protobuf.util.Durations;
+import io.temporal.activity.ActivityCancellationType;
+import io.temporal.activity.ActivityOptions;
+import io.temporal.activity.LocalActivityOptions;
+import io.temporal.api.common.v1.Payload;
+import io.temporal.api.common.v1.WorkflowExecution;
+import io.temporal.api.enums.v1.ParentClosePolicy;
+import io.temporal.common.RetryOptions;
+import io.temporal.common.VersioningIntent;
+import io.temporal.failure.ActivityFailure;
+import io.temporal.failure.ApplicationFailure;
+import io.temporal.failure.CanceledFailure;
+import io.temporal.failure.ChildWorkflowFailure;
+import io.temporal.workflow.*;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import org.slf4j.Logger;
+
+public class KitchenSinkWorkflowImpl implements KitchenSinkWorkflow {
+  public static final Logger log = Workflow.getLogger(KitchenSinkWorkflowImpl.class);
+  Map<String, String> state = new HashMap<>();
+  WorkflowQueue<KitchenSink.ActionSet> signalActionQueue = Workflow.newWorkflowQueue(100);
```

Comment:
How was 100 chosen and what are the consequences of this value?

Reply:
The only consequence if we hit the queue limit is the signal handler threads will block on inserting